site/en/guides/porting/define-cmake-rules.md

```diff
@@ -0,0 +1,135 @@
+# Define CMake rules
+
+The CMake build system relies on `CMakeLists.txt` files which define build targets. The `CMakeLists.txt` at the root of your repository is the top of the build tree and is a good place to start defining the various targets, options, and macros which will be used in the build process.
+
+The very first values to define are the project name and the supported platforms.
+
+In this example, we set the project name to `ot-efr32` with a version of `0.0.1`. We also define a variable `EFR32_PLATFORM_VALUES` which is a list of `efr32` platforms supported by `ot-efr32`. For the sake of this example, we've defined multiple platforms, but having a single platform for the `_PLATFORM_VALUES` variable is fine as well.
+
+```cmake
+cmake_minimum_required(VERSION 3.10.2)
+project(ot-efr32 VERSION 0.0.1)
+
+set(EFR32_PLATFORM_VALUES
+    "efr32mg1"
+    "efr32mg12"
+    "efr32mg13"
+    "efr32mg21"
+)
+```
+
+The `CMakeLists.txt` file includes a check that will prevent a build launched for an unsupported platform.
+```cmake
+set_property(CACHE EFR32_PLATFORM PROPERTY STRINGS ${EFR32_PLATFORM_VALUES})
+if(NOT EFR32_PLATFORM IN_LIST EFR32_PLATFORM_VALUES)
+    message(FATAL_ERROR "Please select a supported platform: ${EFR32_PLATFORM_VALUES}")
+endif()
+```
+
+The next variable which needs to be defined is `OT_PLATFORM_LIB`. This variable is used by the OpenThread example applications to link against your platform.
+
+```cmake
+set(OT_PLATFORM_LIB "openthread-${EFR32_PLATFORM}")
+```
+
+
+## OpenThread CMake options
```

Comment:
Please add "Step 1:" "Step 2:" etc to each level 2 header on this page so it's consistent with the other pages of the porting guide.